### ISSUE TITLE
Mac Installer does not work in Mojave and prior systems.

### ISSUE THREAD

**Issue by hairydalek** [orangeadam3/terra121#134]
Still on Mojave here for various reasons, and the Mac installer does not work on that platform. When you double click on it (and the icon has the usual greyed out overlay, so you should know it won’t work) you get told that it’s not compatible.

There is a workaround:
1 - Right-click on the application bundle and select Show Contents
2 -Open the Contents folder, and then the MacOS Folder.
3 - Double-click on the bteinstaller file
4 >> The installer will open Terminal and run from there. The installer window will open, and you’ll be able to proceed.

**Comment by orangeadam3** [comment]
This is not an issue with the mod itself. I have virtually no input on the launcher development. Please direct bte (launcher, modpack, website, discord) issues to their team.

**Comment by hairydalek** [comment]
OK, thanks.

**Comment by Barteks2x** [comment]
Interesting, so the binary actually works. This means that mac version doesn't like something about the .app structure I made but I have no way to know by myself what.

**Comment by hairydalek** [comment]
Yes - and I have tried tinkering with the .plist in your bundle, using settings from some of my own apps (note: not a full time developer, but I have been known to tinker out of curiosity). I even created a new app in Xcode and tried to get that to run from a bundle.

Each time, I am told that the app needed 10.15 or later. yet if I double-click the executable, it fires up Terminal and runs! Odd.

The structure of the BTEinstaller seems fine.

In Xcode you can set the base OS for the build. That may be what you need to do (I assume you ate using Xcode). HTH. Sorry I can't be more helpful.

**Comment by Barteks2x** [comment]
I don't use xcode for it. I get a native executable out of `native-image` from graalvm and I package it into a .app and then into .dmg "manually" with a few commands in github actions. I was unable to find good documentation on how to do this stuff.

**Comment by hairydalek** [comment]
@Barteks2x I have cobbled together (this is a disguised warning - this is not my day job!) a small App in Xcode that runs your installer. It seems to be running on 10.14.6 here - I let it run an install cycle and it worked. However, test on a number ion platforms.

In Xcode, I have set the minimum OSX to 10.13.

So, I guess there are two paths here:
1 - Use Xcode and change all the project settings so it’s your details in the project! Swap out the BTEInstaller for the one you have created. This is the preferred way of doing things. To create a build to distribute in Xcode, go to Product>Archive, Click the Distribute App button and then choose the Copy App. That will save the version of the app in a folder with your Xcode project file.

2 - Modify the app bundle by changing the BTEInstaller for any updates you make. That will be in Contents/Resources (NOT the MacOS folder!). This may/may not work (depending on how Xcode has built the app - there are security things in place that may hinder apps running that have been modified in this way). Edit the plist file to taste.

Download a zip of the project here. Includes an app, and the Xcode project.
https://www.dropbox.com/s/yzmuhqseq5ht9e9/BTEInstaller.zip?dl=0

Let me know when you have the file so I can remove it from my Dropbox Folder.

Hope this helps your efforts!

**Comment by Barteks2x** [comment]
I think the part that xcode adds that actually matters is
```
	<key>LSMinimumSystemVersion</key>
	<string>10.13</string>
```

I would definitely be using xcode if I actually owned a mac. I don't. All of my setup is done by an educated guess of macos commands that I put into github actions build. I will have to possibly look for information on commandline tools for xcode and how to use all of that. Which is not helped by the fact that I just don't have a mac.

**Comment by hairydalek** [comment]
I tried that key/string pair on its own in the installer I downloaded, and
no dice. I think there’s more going on with the folder structure and files
included in the build.

**Comment by Barteks2x** [comment]
I have no idea what it is and I'm not comfortable releasing dmg I didn't make myself to the public for use, specifically because it contains that weird binary. So unless I find a way to:
 * use xcode from commandline on github actions
 * or, do what xcode is doing manually
 * The binary xcode adds is not needed and can be directly replaced with the one I make
I probably won't do that.

**Comment by jbmagination** [comment]
@hairydalek I wouldn't assume that it would have to be on it's own, but with everything else.
